daemon/fqdn.go

```diff
@@ -121,9 +121,9 @@ func (d *Daemon) updateSelectorCacheFQDNs(selectors map[policyApi.FQDNSelector][
 	// There may be nothing to update - in this case, we exit and do not need
 	// to trigger policy updates for all endpoints.
 	if len(selectors) == 0 && len(selectorsWithoutIPs) == 0 {
-		return
+		return 0
 	}
-	d.TriggerPolicyUpdates(false, "updated identities for FQDNs")
+	return d.TriggerPolicyUpdates(false, "updated identities for FQDNs")
```

Comment:
For this PR to work this needs to be `true`. This has big performance implications, though, as this means we will be regenerating all endpoints quite frequently.

Reply:
We actually have logic which updates the in-memory policymap state for Endpoints. See `ConsumeMapChanges` - in `pkg/policy`.

I think that we shouldn't have to call `TriggerPolicyUpdates` here at all. Instead, we should have a separate type of event which updates the policy maps for all endpoints on the node. This could be an `eventqueue.Event` that we `Enqueue` onto the `EventQueue` for the endpoint. Given that there is no actual change to the policy "rules" with these FQDN updates, we wouldn't need to bump the endpoint's policy revision.

I'm recommending using the EventQueue construct here out of caution, as that guarantees that any regenerations which are in line for the endpoint are taken care of first as well. I'm wondering whether we can simply asynchronously update the policymap state for the endpoint and return a channel when all updates for Endpoints are done.

Reply:
I'm fine with anything, really, as long as we can block appropriately. I guess all FQDN updates of this sort would go through the newer policymap update pipeline so we don't need to mix that and another wait for when a new policy comes in (that wouldn't be a real-time thing anyway).